workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/DashboardTable/DashboardTable.tsx

```diff
@@ -18,50 +19,88 @@ export type AppProps = {
 };
 interface DashboardTableProps {
 	apps: AppProps[];
+	loading: boolean
 }
 
-export function DashboardTable({apps}: DashboardTableProps) {
-	return (
-		<ClayTable borderless className="dashboard-table-container">
-			<ClayTable.Head>
-				<ClayTable.Cell headingCell>
-					<div className="dashboard-table-header-name">
-						<span className="dashboard-table-header-text">
-							Name
-						</span>
-
+export function DashboardTable({apps, loading}: DashboardTableProps) {
+	if (apps.length === 0) {
+		if (loading) {
+			return (
+				<div className="dashboard-table-loading-container">
+					<div className="dashboard-table-loading-image-container">
 						<img
-							alt="Swap Vert"
-							className="dashboard-table-header-name-icon"
-							src={swapVert}
+							alt="Dashboard Loading Image"
+							className="dashboard-table-loading-image"
+							src={dashboardLoadingIcon}
 						/>
 					</div>
-				</ClayTable.Cell>
+					<div className="dashboard-table-loading-text">
+						Apps Loading . . .
+					</div>
+				</div>
+			)
+		} else {
+			return (
+				<div className="dashboard-table-loading-container">
+					<div className="dashboard-table-loading-image-container">
+						<img
+							alt="Dashboard Loading Image"
+							className="dashboard-table-loading-image"
+							src={dashboardLoadingIcon}
+						/>
+						</div>
+					<div className="dashboard-table-no-apps-text">
+						No apps yet
+					</div>
+					<div className="dashboard-table-no-apps-description">
+						Create new apps and they will show up here. Click on "<a href="/create-new-app">New App</a>" to start creating apps
+					</div>
+				</div>
+			)
+		}
+	} else {
+		return (
+			<ClayTable borderless className="dashboard-table-container">
+				<ClayTable.Head>
+					<ClayTable.Cell headingCell>
+						<div className="dashboard-table-header-name">
+							<span className="dashboard-table-header-text">
+								Name
+							</span>
+
+							<img
+								alt="Swap Vert"
+								className="dashboard-table-header-name-icon"
+								src={swapVert}
+							/>
+						</div>
+					</ClayTable.Cell>
```

Comment:
"return" and some markup are repeated in the if/else statement.  Could you try this pattern instead?
`{label && (
	<label className={classNames({disabled})} htmlFor={id}>
		{label}
		{required && <RequiredMask />}
	</label>
)}`.

Reply:
https://github.com/liferay/liferay-portal/blob/0780a959547d962bf35d7c093b087e5ca0ef3677/workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/FieldBase.tsx#L83

Reply:
Should we be loading the whole dashboard page instead of just the dashboard table?  We also have to load the dashboard navigation items as well.